### PR TITLE
fix: fixed the issue with populating checkLinkedDomain value

### PR DESCRIPTION
**Pull request by sksadjad**
- fixed the issue with populating checkLinkedDomain value for verifying jwt
- some changes in the validateLinkedDomainWithDid function

### CODE REVIEW

File: src/main/types/SIOP.types.ts [hunk 1]
```diff
@@ -452,30 +452,30 @@ export enum VerificationMode {
...
+
+export interface ExternalVerification extends Verification {
   verifyUri: string; // url to call to verify the id_token signature
   authZToken?: string; // Optional: bearer token to use to the call
   resolveOpts: ResolveOpts;
 }
 
 export interface VerifyAuthenticationRequestOpts {
-  verification: InternalVerification | ExternalVerification; // To use internal verification or external hosted verification
+  verification: Verification; // To use internal verification or external hosted verification
```
Comment (nklomp) [inline src/main/types/SIOP.types.ts:470]: Don't change this to the super interface. Right now this interface is exactly the same as the InternalVerification. But if we add one unique prop to InternalVerification you would still be able to pass in objects with to few params, as long as they have all props from Verification. It really needs to be either Internal or External Verification

File: src/main/types/SIOP.types.ts [hunk 2]
```diff
@@ -452,30 +452,30 @@ export enum VerificationMode {
...
   checkLinkedDomain?: CheckLinkedDomain;
   // didDocument?: DIDDocument; // If not provided the DID document will be resolved from the request
   nonce?: string; // If provided the nonce in the request needs to match
   // redirectUri?: string;
 }
 
 export interface VerifyAuthenticationResponseOpts {
-  verification: InternalVerification | ExternalVerification;
+  verification: Verification;
```
Comment (nklomp) [inline src/main/types/SIOP.types.ts:478]: Don't change this to the super interface. Right now this interface is exactly the same as the InternalVerification. But if we add one unique prop to InternalVerification you would still be able to pass in objects with to few params, as long as they have all props from Verification. It really needs to be either Internal or External Verification

File: src/main/types/SIOP.types.ts [hunk 3]
```diff
@@ -637,9 +637,9 @@ export const isRequestPayload = (object: AuthenticationRequestPayload | Authenti
 export const isResponsePayload = (object: AuthenticationRequestPayload | AuthenticationResponsePayload): object is AuthenticationResponsePayload =>
   'iss' in object && 'aud' in object;
 
-export const isInternalVerification = (object: InternalVerification | ExternalVerification): object is InternalVerification =>
+export const isInternalVerification = (object: Verification): object is InternalVerification =>
```
Comment (nklomp) [inline src/main/types/SIOP.types.ts:640]: Don't change this to the super interface. Right now this interface is exactly the same as the InternalVerification. But if we add one unique prop to InternalVerification you would still be able to pass in objects with to few params, as long as they have all props from Verification. It really needs to be either Internal or External Verification

File: src/main/types/SIOP.types.ts [hunk 4]
```diff
@@ -637,9 +637,9 @@ export const isRequestPayload = (object: AuthenticationRequestPayload | Authenti
...
   object.mode === VerificationMode.INTERNAL; /* && !isExternalVerification(object)*/
-export const isExternalVerification = (object: InternalVerification | ExternalVerification): object is ExternalVerification =>
+export const isExternalVerification = (object: Verification): object is ExternalVerification =>
```
Comment (nklomp) [inline src/main/types/SIOP.types.ts:642]: Don't change this to the super interface. Right now this interface is exactly the same as the InternalVerification. But if we add one unique prop to InternalVerification you would still be able to pass in objects with to few params, as long as they have all props from Verification. It really needs to be either Internal or External Verification

**Review by nklomp — CHANGES_REQUESTED** [review]
See remark about not using the super interface as argument

File: src/main/types/SIOP.types.ts (hunk 1 above)
Comment (sksadjad) [reply src/main/types/SIOP.types.ts]: reverted to `InternalVerification | ExternalVerification`

File: src/main/types/SIOP.types.ts (hunk 2 above)
Comment (sksadjad) [reply src/main/types/SIOP.types.ts]: reverted to `InternalVerification | ExternalVerification`

File: src/main/types/SIOP.types.ts (hunk 3 above)
Comment (sksadjad) [reply src/main/types/SIOP.types.ts]: reverted to `InternalVerification | ExternalVerification`

File: src/main/types/SIOP.types.ts (hunk 4 above)
Comment (sksadjad) [reply src/main/types/SIOP.types.ts]: reverted to `InternalVerification | ExternalVerification`

**Review by nklomp — APPROVED** [review]

**Pull request merged**